mailpoet/lib/Automation/Engine/Builder/UpdateStepsController.php

```
@@ -21,12 +21,38 @@ public function updateSteps(Automation $automation, array $data): Automation {
     $steps = [];
     foreach ($data as $index => $stepData) {
       $step = $this->processStep($stepData, $automation->getStep($stepData['id']));
-      $steps[$index] = $step;
+      $updatedStep = $this->maybeRunOnDuplicate($step, $automation);
+      $steps[$index] = $updatedStep;
     }
     $automation->setSteps($steps);
     return $automation;
   }
 
+  private function maybeRunOnDuplicate(Step $step, Automation $automation): Step {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$automation'. (Unused Code Rules)

(UnusedFormalParameter)